### ISSUE TITLE
Port of models from Keeling and Rohani

### ISSUE THREAD

**Issue by sdwfrost** [epirecipes/shopping-list#18]
See [http://www.modelinginfectiousdiseases.org/](http://www.modelinginfectiousdiseases.org/)

**Comment by LloydChapman** [comment]
I'd like to implement the SEIR models from Keeling and Rohani and an SEIR version of the spatial kernel model in chapter 7, with an extension for households, in R, Matlab (Octave) and Julia.

**Comment by t-pollington** [comment]
Chapter 4. Section 2.2 Vectored transmission, program 4.4. It's a multi-host SIR model in R(using desolve) as I'm interested in vector-transmitted diseases.
**DONE FOR R VERSION: Find in shared/Multi-host SIR**

**Comment by t-pollington** [comment]
Found an online version of the Keeling & Rohani book here http://acces.ens-lyon.fr/acces/thematiques/sante/epidemiologie/GlossairEpidem/livre_K-R.pdf

**Comment by theresasophia** [comment]
I am interested in implementing models for host heterogeneities from Chapter 3 in R

**Comment by emmaaccorsi** [comment]
I'm also interested in working on the models from Chapter 3 in R, in particular 3.1 and 3.2. Are you planning to implement all of Chapter 3?

**Comment by rek160** [comment]
I am interested in implementing models from Chapter 8 in Julia

**Comment by iholmdahl** [comment]
I'm interested in the metapopulation models for animals and humans from chapter 7

**Comment by theresasophia** [comment]
> I'm also interested in working on the models from Chapter 3 in R, in particular 3.1 and 3.2. Are you planning to implement all of Chapter 3?

No, I am interested in the age-structured models, so I could do 3.3 and/or 3.4.

**Comment by emmaaccorsi** [comment]
> > I'm also interested in working on the models from Chapter 3 in R, in particular 3.1 and 3.2. Are you planning to implement all of Chapter 3?
> 
> No, I am interested in the age-structured models, so I could do 3.3 and/or 3.4.

Perfect!

**Comment by lvcooper** [comment]
I'm interested in working on multi-strain/multi-pathogen models from Chapter 4.1 in R.

**Comment by iholmdahl** [comment]
> I'm interested in the metapopulation models for animals and humans from chapter 7

Actually, switching for now to seasonal forcing from chapter 5

**Comment by eqmooring** [comment]
I am interested in, at least in the first instance, writing up 7.1 and 7.2 (basic spatial models) in R.

**Comment by eqmooring** [comment]
> I'd like to implement the SEIR models from Keeling and Rohani and an SEIR version of the spatial kernel model in chapter 7, with an extension for households, in R, Matlab (Octave) and Julia.


Sounds good. I will initially focus on earlier models in Chapter 7.

**Comment by LloydChapman** [comment]
> > I'd like to implement the SEIR models from Keeling and Rohani and an SEIR version of the spatial kernel model in chapter 7, with an extension for households, in R, Matlab (Octave) and Julia.
> 
> Sounds good. I will initially focus on earlier models in Chapter 7.

I've decided to change tack and focus on implementing an SEIR version of the spatial kernel model in 7.5 and extended versions. I have however implemented the basic SEIR model in Chapter 2 in R and Julia.

**Comment by EBucksJeff** [comment]
I'm going to start working on 3.5 in R, if anyone isn't already?

**Comment by aineniamh** [comment]
I'm working through implementing models from Chapter 2 in python using widgets.

**Comment by rek160** [comment]
> I am interested in implementing models from Chapter 8 in Julia

Have done 8.1 and 8.2 in Julia and now doing them in R

**Comment by emmaaccorsi** [comment]
> I'm also interested in working on the models from Chapter 3 in R, in particular 3.1 and 3.2. Are you planning to implement all of Chapter 3?

3.1 and 3.2 are complete in R.

**Comment by sdwfrost** [comment]
Can you put finished models on the server, e.g. in shared/kr08/3_1, shared/kr08/3_2 etc., so I can review and integrate into the cookbook?

**Comment by HelenaStage** [comment]
Interested in 4.3 in Python.

**Comment by Federica-Giardina** [comment]
Starting 6.1 and 6.2 in R if nobody is working on it

**Comment by emmaaccorsi** [comment]
> Can you put finished models on the server, e.g. in shared/kr08/3_1, shared/kr08/3_2 etc., so I can review and integrate into the cookbook?

Yes, they should be uploaded there now. Thanks!

**Comment by c-lack** [comment]
I'm going to make 7.8 in JavaScript. It'll be on Observable though, so not on the server.

**Comment by t-pollington** [comment]
**QUIRK IN LSODA() FOR R_DESOLVE**

When using the init or parms arguments for lsoda() make sure they are named. In most of the examples they are explicitly named i.e. parms <- c(beta=0.1,gamma=0.05) which doesn't just give 0.1 & 0.05 as information to the ODE function but also gives the column labels "beta" and "gamma". Doing parms = c(beta,gamma) isn't sufficient as it won't inherit the variable labels. To fix this either:
i) 
After doing parms = c(0.1,0.05)
do names(parms) = c("beta","gamma") where the labels match the equation notation in sirODE() function.
ii) 
In sirODE(times, init, parms){
Add some lines that define the labels wrt the parms argument of this function i.e.
beta = parms[1]
gamma = parms[2]

then you can continue with the default:
dS <- -beta*S*I
dI <- beta*S*I-gamma*I
dR <- gamma*I

}

**Comment by emmaaccorsi** [comment]
I'm doing 3.1 and 3.2 in Julia.

**Comment by theresasophia** [comment]
We will implement 3.3 and 3.4 in Julia

**Comment by EBucksJeff** [comment]
3.5 done in R and in the shared folder.

**Comment by ersouthall** [comment]
I'm doing 7.6 (FMD Spatial model) in observable

**Comment by nayantaraw** [comment]
I am working on 5.1 and 7.2 from Keeling and Rohani in R.

**Comment by emmaaccorsi** [comment]
> I'm doing 3.1 and 3.2 in Julia.

3.1 and 3.2 in Julia are complete and in the shared folder.

**Comment by rek160** [comment]
> > I am interested in implementing models from Chapter 8 in Julia
> 
> Have done 8.1 and 8.2 in Julia and now doing them in R

Also doing 8.1 in Observable

**Comment by iholmdahl** [comment]
I've done 5.1 and 5.4 in R and Julia, now doing in Observable

**Comment by cnd27** [comment]
Working with 6.3 in Julia

**Comment by eqmooring** [comment]
> I am interested in, at least in the first instance, writing up 7.1 and 7.2 (basic spatial models) in R.

7.1 in R using deSolve plus an Intro file with documentation on using matrices for metapopulation calculations are in the shared/kr08 folder. I am now working on 7.1 in Julia.
